### PR TITLE
Tiered cart promo

### PR DESCRIPTION
**Pull request by patryksmolarz**

### CODE REVIEW

File: tiered-cart-promotions/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Tiered cart promotions with Voucherify <a id="voucherify-example"></a>
+
+This sample shows how we can reward our customers by adding rewards for more money spent. We can achieve that solution thanks [validate promotion](https://docs.voucherify.io/reference/validate-promotions-1) endpoint. At the end we should use [redeem stackable](https://docs.voucherify.io/reference/redeem-stacked-discounts) endpoint like in the other examples. The stacking mechanism allows you to combine up to 5 promo codes or cart-level promotions with a single request.
+
+Validating and accepting promo codes in your checkout from scratch might be tricky — calculating discounted prices, error message handling, and localization are just a few things to think about when building a simple promo code redemption flow.
+
+This is where the [Voucherify promotion engine](https://docs.voucherify.io/docs) kicks in. Together with our [Promo UI Kit](https://www.figma.com/community/file/1100356622702326488) you can quickly build the best promotion experience for your customers.
+
+This example introduce an online coffee shop and many vouchers in your dashboard are about it.
```
Comment (akaliszuk) [inline tiered-cart-promotions/README.md]: introduces

File: tiered-cart-promotions/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Demo <a id="demo"></a>
+
+Live demo on:<br>
+[<img src="https://cdn.icon-icons.com/icons2/2699/PNG/512/heroku_logo_icon_169035.png" width="100px"/>](https://voucherify-examples.herokuapp.com/tiered-cart-promotions/)
+
+![](https://github.com/voucherify-samples/voucher-code-redemption/blob/main/free_shipping.gif)
+
+The demo is running with a [Sandbox project](https://docs.voucherify.io/docs/testing). Sandbox comes with test campaign "Reward promotion" you can apply in your example or you can add your own [reward campaign](https://support.voucherify.io/article/519-create-cart-level-promotions).
```
Comment (akaliszuk) [inline tiered-cart-promotions/README.md]: Sandbox comes with test campaign => Sandbox comes with a test campaign

File: tiered-cart-promotions/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+```In this example applying coupons is not possible.```
+
+This sample calls two endpoints:
+
+* [Validate promotion](https://docs.voucherify.io/reference/validate-promotions-1) - check if any [promotion tier](https://docs.voucherify.io/docs/promotion-tier) exists.
+* [Redeem rewards](https://docs.voucherify.io/reference/redeem-voucher) — runs validation and then marks the rewards as used. After clicking the redemption button you should see the message "Rewards redeemed" - that means your redemption process was successfull.
+
```
Comment (akaliszuk) [inline tiered-cart-promotions/README.md:33]: successful

File: tiered-cart-promotions/client/js/lib.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    const summedDiscountPrice = rewards?.reduce((sum, voucher) => sum + voucher.discount, 0) / 100;
+    const subtotal = sumProductPrices(products);
+    const discountValue = discountsTemplate.querySelector(".discount-value span").innerHTML = `$${summedDiscountPrice || 0}`;
+    discountsTemplate.querySelector(".all-discounts span").innerHTML = `$${summedDiscountPrice || 0}`;
+    discountsTemplate.querySelector(".subtotal span").innerHTML = `$${subtotal}`;
+    rewards?.map(reward => {
+        discountsTemplate.querySelector(".coupons").innerHTML += `<h5 class="coupon"><span class="coupon-value">Reward Tier ${reward.hierarchy}</span></h5>`;
+    });
+    const shipping = discountsTemplate.querySelector(".shipping span").innerHTML = "$8.99";
+    discountsTemplate.querySelector(".grand-total span").innerHTML = `$${(+shipping.replace("$", "") + +subtotal - discountValue.replace("$", "")).toFixed(2)}`;
```
Comment (akaliszuk) [inline tiered-cart-promotions/client/js/lib.js]: Maybe simplify this case? Assign shipping amount to variable. '+' stands for parsing to number this values here?

**Review by akaliszuk — CHANGES_REQUESTED** [review]
Review

File: tiered-cart-promotions/client/js/lib.js (hunk 4 above)
Comment (patryksmolarz) [reply tiered-cart-promotions/client/js/lib.js]: yep, + mean that it's a number

**Review by akaliszuk — APPROVED** [review]

**Pull request merged**